TidepoolOnboardingKitUI/Support/OnboardingSection.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  OnboardingSection.swift
+//  TidepoolOnboardingKitUI
+//
+//  Created by Darin Krauss on 3/4/21.
+//  Copyright © 2021 Tidepool Project. All rights reserved.
+//
+
+/// The onboarding sections, in order of progression
+enum OnboardingSection: String, CaseIterable, Codable, Equatable {
+    case welcome
+    case introduction
+    case howTheAppWorks
+    case aDayInTheLife
+    case yourSettings
+    case yourDevices
+    case getLooping
+
+    enum State: Equatable {
+        case completed
+        case available
+        case unavailable
+    }
+
+    enum Progress: String, Codable, Equatable {
+        case started
+        case completed
+    }
+}
+
+struct OnboardingSectionProgress: RawRepresentable, Codable, Equatable {
+    typealias RawValue = [String: Any]
+
+    let section: OnboardingSection
+    let progress: OnboardingSection.Progress
+    let date: Date
+    let timeZone: String
```

Comment:
Eventually (a later ticket in Jira) this information will make its way up to the Tidepool backend so we retain data confirming the user went through all steps of onboarding. Product asked to capture the start and completion time of each section, partially for this confirmation and partially for future validation that our section time estimates are accurate. This captures that information. The time zone is added for completeness since the encoded `Date` is GMT (+0:00). It is captured as a `String` to eliminate the error case on rehydration of the `TimeZone` where `init?(identifier: String)` returns `nil`. Besides, that is how it will be sent to the backend in JSON.

Going to keep this here as a `String` for simplicity.